consensus.go

```diff
@@ -178,6 +179,10 @@ type Pbft struct {
 
 	// notifier is a reference to the struct which encapsulates handling messages and timeouts
 	notifier StateNotifier
+
+	//for experimenting
+	roundTimeoutFunc  func() <-chan time.Time
+	proposalDelayFunc func(ctx context.Context, delay time.Duration)
```

Comment:
Just to simplify tests. I don't need to think about proposal time for writing tests.

Reply:
But could we get the same result just by setting the delay in the block proposal? Then, I would be on favour of using that instead of another test wrapper.